storage/connect/tabmul.cpp

```diff
@@ -567,11 +573,12 @@ bool TDBMSD::InitFileNames(PGLOBAL g)
 	while (true)
```

Comment:
I useful check before this point would be a check if `filename == NULL`. Also obscured by obsessive overloading here is that `FNSZ + 1` bytes is allocated.

Reply:
This not a part of my change but anyway I can add the check here. The question is - should we really do this?
`filename` can be  `NULL` here in case of `PlugSubAlloc()` fails but it should write error message and thow an exception then - see https://github.com/MariaDB/server/blob/10.4/storage/connect/plugutil.cpp#L581-L587

I agree it's not very clear here. I can add additional check if you still think it's right thing to do.